packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/useHandleUpload.ts

```diff
@@ -26,10 +26,18 @@ interface TaskUpdate extends Partial<CancelableTask> {
 
 const removeTask = <T extends Task | CancelableTask>(
   tasks: T[],
-  key: string
+  key: string,
+  updateFiles: React.Dispatch<React.SetStateAction<File[]>>
```

Comment:
Running a side effect from a consumer can lead to subtle complexity as useHandleUpload is no longer a pure function, relying on the behavior of an external module. Ideally we should update useHandleUpload to expose a handler that can be provided to useFileSelect that will accept files and convert them to tasks/merge with existing tasks to ensure a single source of truth and simplify task lifecycle management